connector.go

```diff
@@ -202,6 +192,24 @@ func (wrb *WapiRequestBuilder) BuildBody(obj IBObject) []byte {
 		return nil
 	}
 
+	if t == GET && len(eaSearch) > 0 {
+		payload := make(map[string]string)
+		json.Unmarshal(jsonStr, &payload)
+		for k, v := range eaSearch {
+			str, ok := v.(string)
+			if !ok {
+				log.Printf("Cannot marshal EA Search attribute for '%s'\n", k)
+			} else {
+				payload["*"+k] = str
+			}
+		}
+		jsonStr, err = json.Marshal(payload)
+		if err != nil {
+			log.Printf("Cannot marshal EA's in the payload: '%s'", payload)
+			return nil
+		}
+	}
+
```

Comment:
@ishantt , thank you for digging into it. I just assumed that what I proposed would work - my bad.

Since the golang proposal has been kicked around since 2013, I doubt that it will be adopted anytime soon - it has already been rejected for 1.9.

In the meantime, what do you think about something like this: https://play.golang.org/p/Qay9YPCv6O

Thanks!